src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitSubscriber.java

```diff
@@ -169,12 +172,46 @@ private void handle(String consumeTag, Delivery delivery) {
         }
     }
 
+    private boolean resubscribe() {
+        LOGGER.info("Trying to resubscribe to queue.");
+
+        RabbitMQConfiguration configuration = this.connectionManager.get().getConfiguration();
+        int attemptsCount = 0;
+        int amountOfAttempts = configuration.getMaxRecoveryAttempts();
+        boolean successful = false;
+
+        while (attemptsCount < amountOfAttempts) {
+            try {
+                Thread.sleep(configuration.getMaxConnectionRecoveryTimeout());
+            } catch (InterruptedException | IllegalArgumentException e) {
+                LOGGER.error(e.getMessage());
+            }
+
+            try {
+                attemptsCount++;
+                consumerMonitor.set(null);
+                start();
+
+                CommonMetrics.setReadiness(true);
```

Comment:
We should register the specific monitor for this case instead of using the user monitor